src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -63,13 +63,42 @@ public struct MetricPoint
             {
                 this.histogramBuckets = null;
             }
+
+            this.EndTime = default;
+            this.StartTime = startTime;
         }
 
         /// <summary>
         /// Gets the tags associated with the metric point.
         /// </summary>
         public ReadOnlyTagCollection Tags { get; }
 
+        public DateTimeOffset StartTime
```

Comment:
Will this be aggressively inlined? Does it take a copy?

Reply:
I don't think we can add AggresiveInlining attribute to a property, so no way to force it.

Reply:
> Does it take a copy?

I'm supposing yes. Though looks like `DateTimeOffset` is basically just [a `short` and a `DateTime`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/DateTimeOffset.cs#L64-L65) where a `DateTime` [is a `ulong`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L143)

Reply:
> I don't think we can add AggresiveInlining attribute to a property, so no way to force it.

Does this work?

```csharp
[MethodImpl(MethodImplOptions.AggressiveInlining)]
get
{
    return this.something;
}
```

Reply:
@reyang Thanks for the suggestion! I have added inlining as well now.

Reply:
The 2nd part of the question is "does it take a copy", do you see a perf penalty of returning a copy, or asking it in another way - it is possible/beneficial to return a ref?

Reply:
There were no targeted benchmarks against `DateTimeOffset`, so can't say if its significant. 
The `MetricPoint` is a big enough struct to avoid copy, so it the `MetricPointAccessor` uses ref to return `MetricPoint` .

Reply:
Run the [Prometheus serialization benchmark](https://github.com/open-telemetry/opentelemetry-dotnet/blob/a828aa71cbe61f2c7fc8727619a6f172867cda38/test/Benchmarks/Exporter/PrometheusSerializerBenchmarks.cs) (which in turn access the timestamps) to see if there is a noticeable diff.

Reply:
Given DateTimeOffset is not big structure, don't think it'll be an issue.
Its possible to add a *new* method returning ref, if this turns out to be important enough to avoid the copy.

Reply:
> 

Prometheus doesn't access the Start/EndTime!

Reply:
IIRC it uses the end time?

Reply:
tried a benchmark which returns StartTime as DateTimeOffset (current PR), vs a long (unix ns representation of the DateTimeOffset), Very minute difference in the read perf.

Given this, we should be good to StartTime as DateTimeOffset now. We may add a *new* property StartTimeUnixNanoSeconds in the future returning `long` for *slight* perf gain. (The Metric data model uses unixtimenanosecs for starttime,endtime, so we could make that as the only exposed property as well, but given we are close to release, making a last minute change is not desired.)

Reply:
|            Method |      Mean |     Error |    StdDev |    Median | Allocated |
|------------------ |----------:|----------:|----------:|----------:|----------:|
|     DateTimeReads | 0.0583 ns | 0.0275 ns | 0.0745 ns | 0.0255 ns |         - |
| DateTimeReadsUnix | 0.0540 ns | 0.0276 ns | 0.0795 ns | 0.0063 ns |         - |

Reply:
```
// <copyright file="MetricPointBenchmarks.cs" company="OpenTelemetry Authors">
// Copyright The OpenTelemetry Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// </copyright>

using System;
using System.Threading;
using BenchmarkDotNet.Attributes;

/*
BenchmarkDotNet=v0.13.1, OS=Windows 10.0.22000
Intel Core i7-8650U CPU 1.90GHz (Kaby Lake R), 1 CPU, 8 logical and 4 physical cores
.NET SDK=6.0.100
  [Host]     : .NET Core 3.1.21 (CoreCLR 4.700.21.51404, CoreFX 4.700.21.51508), X64 RyuJIT
  DefaultJob : .NET Core 3.1.21 (CoreCLR 4.700.21.51404, CoreFX 4.700.21.51508), X64 RyuJIT


|                   Method |      Mean |     Error |    StdDev |    Median | Allocated |
|------------------------- |----------:|----------:|----------:|----------:|----------:|
|      UpdateStructInArray | 0.6951 ns | 0.1085 ns | 0.3078 ns | 0.5903 ns |         - |
|       UpdateClassInArray | 1.4135 ns | 0.0767 ns | 0.1477 ns | 1.3856 ns |         - |
| UpdateSealedClassInArray | 1.3502 ns | 0.0787 ns | 0.2284 ns | 1.2962 ns |         - |
*/

namespace Benchmarks.Metrics
{
    [MemoryDiagnoser]
    public class MetricPointBenchmarks
    {
        private MetricPointStruct metricpoint;
        private DateTimeOffset start;
        private DateTimeOffset end;
        private long startUnix;
        private long endUnix;

        [GlobalSetup]
        public void Setup()
        {
            this.metricpoint = default(MetricPointStruct);
        }

        [GlobalCleanup]
        public void Cleanup()
        {
        }

        [Benchmark]
        public void DateTimeReads()
        {
            this.start = this.metricpoint.StartTime;
            this.end = this.metricpoint.EndTime;
        }

        [Benchmark]
        public void DateTimeReadsUnix()
        {
            this.startUnix = this.metricpoint.StartTimeUnixNanoSeconds;
            this.endUnix = this.metricpoint.EndTimeUnixNanoSeconds;
        }
    }

#pragma warning disable SA1402 // File may only contain a single type
#pragma warning disable SA1201 // Elements should appear in the correct order
    internal struct MetricPointStruct
    {
        private DateTimeOffset startTime;
        private DateTimeOffset endTime;
        private long startUnix;
        private long endUnix;

        public MetricPointStruct()
        {
            this.startTime = DateTimeOffset.Now;
            this.endTime = DateTimeOffset.Now;
            this.startUnix = this.startTime.ToUnixTimeNanoseconds();
            this.endUnix = this.endTime.ToUnixTimeNanoseconds();
        }

        public DateTimeOffset StartTime
        {
            get
            {
                return this.startTime;
            }
        }

        public DateTimeOffset EndTime
        {
            get
            {
                return this.endTime;
            }
        }

        public long StartTimeUnixNanoSeconds
        {
            get
            {
                return this.startUnix;
            }
        }

        public long EndTimeUnixNanoSeconds
        {
            get
            {
                return this.endUnix;
            }
        }
    }

    internal static class TimestampHelpers
    {
        private const long NanosecondsPerTicks = 100;
        private const long UnixEpochTicks = 621355968000000000; // = DateTimeOffset.FromUnixTimeMilliseconds(0).Ticks

        internal static long ToUnixTimeNanoseconds(this DateTimeOffset dto)
        {
            return (dto.Ticks - UnixEpochTicks) * NanosecondsPerTicks;
        }
    }
#pragma warning restore SA1201 // Elements should appear in the correct order
#pragma warning restore SA1402 // File may only contain a single type
}

```